### ISSUE TITLE
Is loading a ktx2 file from a .glb supported?

### ISSUE THREAD

**Issue by DerrickBarra** [atteneder/KtxUnity#29]
@atteneder Hey bud, our team just added KtxUnity plugin integration to the GLTFUtility plugin (we would use GLTFast, but we need animation support).

However in my testing this morning, KtxUnity has been returning a null reference to the texture on the success delegate. If I load a .ktx2 or .basis texture outside of a .glb there aren't any issues.

The weird thing is that the mimeType is confirmed to be image/ktx2 beforehand.

Now the issue could also be with the optimizer we're passing our models through (Rapid Compact from DGG) and the creation of the ktx2. But can you confirm you've seen this test work? And if so, can you share the test model you used so I can try it out here?

Here's the file I'm testing if you're curious.

Horse
[GLB](https://bxrtest.s3.amazonaws.com/Draco_And_KTX_Testing/horse_NoKTX2_NoDraco.glb)
[GLB - KTX2](https://bxrtest.s3.amazonaws.com/Draco_And_KTX_Testing/horse_KTX2_NoDraco.glb)
[GLB - Draco](https://bxrtest.s3.amazonaws.com/Draco_And_KTX_Testing/horse_NoKTX2_DracoLossy.glb)
[GLB - Draco & KTX2](https://bxrtest.s3.amazonaws.com/Draco_And_KTX_Testing/horse_KTX2_DracoLossy.glb)

![GLTFImage](https://user-images.githubusercontent.com/20777145/98850960-8ba16e80-2423-11eb-8439-f97e243592fb.PNG)

**Comment by atteneder** [comment]
Hi @DerrickBarra ,

Transcoding likely failed.

Last time I checked RapidCompact (4.0.x), they seem to still use an outdated KTX 2.0 draft specification. Not sure, but could be the reason.

I successfully tested with [gltfpack 0.15](https://github.com/zeux/meshoptimizer/tree/master/gltf) and suggest you try it as well to compare. If this works, it's likely RapidCompact's fault.

Let me know how it goes.

**Comment by DerrickBarra** [comment]
@atteneder I just reached out with Rapid Compact to double check that the CLI we are using has the latest KTX specification. I'm pretty sure they updated to the latest a few months ago but maybe our CLI is out of date.I'll investigate that tool you linked me to which will help us diagnose the problem. Thanks!Sent from my Galaxy
-------- Original message --------From: Andreas Atteneder <notifications@github.com> Date: 11/11/20  5:13 PM  (GMT-05:00) To: atteneder/KtxUnity <KtxUnity@noreply.github.com> Cc: Derrick Barra <derrickbarra@brandvr.co>, Mention <mention@noreply.github.com> Subject: Re: [atteneder/KtxUnity] Is loading a ktx2 file from a .glb supported? (#29) 
Hi @DerrickBarra ,
Transcoding likely failed.
Last time I checked RapidCompact (4.0.x), they seem to still use an outdated KTX 2.0 draft specification. Not sure, but could be the reason.
I successfully tested with gltfpack 0.15 and suggest you try it as well to compare. If this works, it's likely RapidCompact's fault.
Let me know how it goes.

**Comment by DerrickBarra** [comment]
@atteneder So I pinpointed the bug in KtxUnity that is causing the issue.

If I use TextureBase.LoadBytesRoutine(), that causes the texture to be returned null. This is a nice convenience function to have around since we're not always working without a monobehaviour, hopefully, the bug in it can get squashed.

Instead, if I use LoadFromBytes() and pass in a Singleton monobehaviour I created as a dummy, then that works just fine.

In talking to another developer (@daverin over at the GLTFUtility plugin git repo). He mentions that he actually squashed this bug with two different solutions. [Check out his reply](https://github.com/Siccity/GLTFUtility/pull/106#issuecomment-726352423).
